File: src/main/kotlin/stellarwitch7/libstellar/registry/Registrar.kt
package stellarwitch7.libstellar.registry

import net.minecraft.registry.Registry
import net.minecraft.util.Identifier

interface Registrar<T> {
    val modID: String
    val registry: Registry<T>

    fun id(name: String): Identifier {
        return Identifier.of(modID, name)
    }

    fun register() {
    }

    fun <V : T> register(name: String, value: V): V {
        return Registry.register(registry, id(name), value)
    }
}

File: src/main/scala/stellarwitch7/libstellar/registry/Registrar.scala
package stellarwitch7.libstellar.registry

import net.minecraft.registry.Registry
import net.minecraft.util.Identifier

trait Registrar[T] {
  val modID: String
  val registry: Registry[T]

  def id(name: String): Identifier = Identifier.of(modID, name)
  def register(): Unit = { }
  def register[V <: T](name: String, value: V): V = Registry.register(registry, id(name), value)
}
